qt/python/mantidqt/project/recovery/recoverygui/projectrecoverymodel.py

```diff
@@ -0,0 +1,179 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantidqt package
+#
+
+from __future__ import (absolute_import, unicode_literals)
+
+import os
+import sys
+from qtpy.QtCore import Slot, QObject
+
+from mantid.api import AnalysisDataService as ADS
+from mantid.kernel import ConfigService
+
+
+class ProjectRecoveryModel(QObject):
+    def __init__(self, project_recovery, presenter):
+        super(ProjectRecoveryModel, self).__init__()
+        self.presenter = presenter
+        self.project_recovery = project_recovery
+
+        self.is_recovery_running = False
+        self.has_failed_run = False
+
+        self.rows = []
+        self.fill_rows()
+
+        self.selected_checkpoint = None
+
+    def find_number_of_workspaces_in_directory(self, path):
+        if not os.path.exists(path):
+            raise AttributeError("Project Recovery Model: Path is not valid")
+        files = self.project_recovery.listdir_fullpath(path)
+        total_counted = 0
+        for file in files:
+            _, file_ext = os.path.splitext(file)
+            if file_ext == ".py":
+                total_counted += 1
+
+        return total_counted
+
+    def get_row(self, checkpoint):
+        # Only check unicode if it's python2
+        if isinstance(checkpoint, str) or (sys.version_info.major < 3 and isinstance(checkpoint, unicode)):
+            # Assume if there is a T then it is a checkpoint and it needs to be replaced with a space
+            checkpoint = checkpoint.replace("T", " ")
```

Comment:
We seem to be doing this replacement in several places. I think this is the case where a `Checkpoint` type would be helpful...